p2p/libp2p/networksharding/peerDistances.go

```diff
@@ -0,0 +1,32 @@
+package networksharding
+
+import (
+	"math/big"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+// peerDistance is a composite struct on top of a peer ID that also contains the kad distance measured
+// against the current peer and held as a big.Int
+type peerDistance struct {
+	peer.ID
+	distance *big.Int
+}
+
+// peerDistances represents a sortable peerDistance slice
+type peerDistances []peerDistance
+
+// Len returns the length of this slice
+func (pd peerDistances) Len() int {
+	return len(pd)
+}
+
+// Less is used in sorting and returns if i-th element is less than j-th element
+func (pd peerDistances) Less(i, j int) bool {
```

Comment:
Less and Swap are needed to be able so apply list.sort. Now I know this :+1:

Reply:
:)

Reply:
ok